packages/widget/src/connect.ts

```diff
@@ -0,0 +1,129 @@
+import { consume } from '@lit/context';
+import { LitElement, html } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+import {
+  WalletManagerContext,
+  WalletManagerController
+} from '@buildwithsygma/sygmaprotocol-wallet-manager';
+
+import {
+  SdkManagerContext,
+  SdkManager
+} from '@buildwithsygma/sygmaprotocol-sdk-manager';
+
+import { Environment } from '@buildwithsygma/sygma-sdk-core';
+
+@customElement('connect-dialog')
+class ConnectDialog extends LitElement {
+  @consume({ context: WalletManagerContext, subscribe: true })
+  @state()
+  walletManager?: WalletManagerController;
+
+  @consume({ context: SdkManagerContext, subscribe: true })
+  @state()
+  sdkManager?: SdkManager;
+
+  @state()
+  chainId?: number;
+
+  async connectedCallback(): Promise<void> {
+    super.connectedCallback();
+    this.chainId = (
+      await this.walletManager?.evmWallet?.web3Provider?.getNetwork()
+    )?.chainId;
+
+    this.walletManager?.addAccountChangedEventListener(() => {
+      this.requestUpdate();
+    });
+    this.walletManager?.addChainChangedEventListener(async () => {
+      this.chainId = (
+        await this.walletManager?.evmWallet?.web3Provider?.getNetwork()
+      )?.chainId;
+
+      this.requestUpdate();
+    });
+  }
+
+  async connect() {
+    await this.walletManager?.connectEvmWallet();
+    this.requestUpdate();
+  }
+
+  async initSdk() {
+    if (!this.walletManager?.evmWallet?.web3Provider) {
+      throw new Error('No provider');
+    }
+    await this.sdkManager?.initialize(
+      this.walletManager.evmWallet?.web3Provider,
+      Environment.TESTNET
+    );
+    this.requestUpdate();
+  }
+
+  async createTransfer() {
+    if (!this.walletManager?.evmWallet?.address) {
+      throw new Error('No wallet connected');
+    }
+    await this.sdkManager?.createTransfer(
+      this.walletManager.evmWallet.address,
+      this.chainId === 11155111 ? 5 : 11155111,
```

Comment:
it'd be good to extract these magic numbers to constants

Reply:
All the code in the `widget` package is entirely out of scope for this PR and will be rewritten to be properly dynamic in due course.